src/Lean/Elab/Binders.lean

```diff
@@ -182,7 +182,7 @@ register_builtin_option checkBinderAnnotations : Bool := {
 /-- Throw an error if `type` is not a valid local instance. -/
 private partial def checkLocalInstanceParameters (type : Expr) : TermElabM Unit := do
   let .forallE n d b bi ← whnf type | return ()
-  if !b.hasLooseBVar 0 then
+  if bi != .instImplicit && !b.hasLooseBVar 0 then
```

Comment:
Could you please add a comment explaining why the new test was added here?

Reply:
Done!